### ISSUE TITLE
[docker] API Gateway  - Impossible to connect ES container

### ISSUE THREAD

**Issue by lmftn** [gravitee-io/issues#3278]
Docker & Gravitee-gateway - Impossible to connect ES container

### Expected Behavior
There is several weird behavior with the Docker gateway image.
Sme variables are not taken into accounts in the gateway.yml file, there is some "localhost" remaining. It's necessary to copy and change the file manually to be able to launch the container and connect to the mongo container.

My problem is not there since I've been able to override the gravitee.yml file.
But, when lauching the container, it tries to connect to my elastic search container on "elastic-search/172.0.0.1:9200". Obvisouly it can't connect to this container since it should be either "elastic-search:9200" or "172.0.0.1:9200".

I'm unable to find where this is called, can you please fix this or tell me where to fix it ? 

Here is my docker-compose file, if needed
<img width="565" alt="Capture d’écran 2020-03-03 à 12 10 43" src="https://user-images.githubusercontent.com/31238854/75770334-061fd780-5d48-11ea-9c82-5d627e554bc0.png">

### Steps to Reproduce (for bugs)

1. Copy the docker-compose provided
2. `docker-compose up`
3. Then the error happens:
`[vert.x-eventloop-thread-3] [] ERROR i.g.e.client.http.HttpClient - Unable to get a connection to Elasticsearch
io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: gravitee-elasticesearch/172.0.0.1:9200
at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)
at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)
at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:327)
at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:340)
at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:665)
at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:612)
at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:529)
at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:491)
at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:905)
at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
at java.lang.Thread.run(Thread.java:748)
Caused by: java.net.ConnectException: Connection refused
... 11 common frames omitted`

### Your Environment

* Version used: Image 1.30.5
* Operating System and version: Docker desktop community v2.2.0.3 / Compose v1.25.4

**Comment by brasseld** [comment]
Hi @lmftn 

Why not defining environment variable for the gateway container to configure connection to ES ?

We did it for the whole platform using our docker compose.
See how it works at https://github.com/gravitee-io/gravitee-docker/blob/master/platform/docker-compose.yml#L71

**Comment by lmftn** [comment]
Hello,

Thanks for the quick reply.
I tried doing this, but I finally configured the endpoint in the gravitee.yml file directly, since there is a remaining "localhost" in the referenced property.
<img width="548" alt="Capture d’écran 2020-03-03 à 15 02 07" src="https://user-images.githubusercontent.com/31238854/75782944-f6ac8880-5d5f-11ea-8f18-0cfa94da5613.png">

Even if I kept `http://${ds.elastic.host}:${ds.elastic.port}` part, it is still trying to call "name of the container/ ip of the container: port of the container". I'm unabled to find where this call is made to change this

**Comment by brasseld** [comment]
Sorry, but where do you see reference to `localhost` ?

**Comment by lmftn** [comment]
In the gravitee.yml file ? That takes all those references.

The error might come from the elastic search image I'm using. I used the official image 7.6 and I'm getting this weird error with the connexion refused. I did try tu uses the same as in your docker-compose file, and it does seem to work.

I don't really get why I have a connexion refused on a weird endpoint ... But problem is fixed for me anyway.
